diff --git a/node_modules/prismarine-viewer/viewer/lib/entity/Entity.js b/node_modules/prismarine-viewer/viewer/lib/entity/Entity.js
index 8945452..dab25be 100644
--- a/node_modules/prismarine-viewer/viewer/lib/entity/Entity.js
+++ b/node_modules/prismarine-viewer/viewer/lib/entity/Entity.js
@@ -203,7 +203,7 @@ function getMesh (texture, jsonModel) {
 class Entity {
   constructor (version, type, scene) {
     const e = entities[type]
-    if (!e) throw new Error(`Unknown entity ${type}`)
+    if (!e) return; //throw new Error(`Unknown entity ${type}`)
 
     this.mesh = new THREE.Object3D()
     for (const [name, jsonModel] of Object.entries(e.geometry)) {
diff --git a/node_modules/prismarine-viewer/viewer/lib/models.js b/node_modules/prismarine-viewer/viewer/lib/models.js
index 9f8938e..c124c9b 100644
--- a/node_modules/prismarine-viewer/viewer/lib/models.js
+++ b/node_modules/prismarine-viewer/viewer/lib/models.js
@@ -130,7 +130,7 @@ function renderLiquid (world, cursor, texture, type, biome, water, attr) {
     if (!neighbor) continue
     if (neighbor.type === type) continue
     if ((neighbor.isCube && !isUp) || neighbor.material === 'plant' || neighbor.getProperties().waterlogged) continue
-    if (neighbor.position.y < 0) continue
+    // if (neighbor.position.y < 0) continue
 
     let tint = [1, 1, 1]
     if (water) {
@@ -241,7 +241,7 @@ function renderElement (world, cursor, element, doAO, attr, globalMatrix, global
       if (!neighbor) continue
       if (cullIfIdentical && neighbor.type === block.type) continue
       if (!neighbor.transparent && neighbor.isCube) continue
-      if (neighbor.position.y < 0) continue
+      // if (neighbor.position.y < 0) continue
     }
 
     const minx = element.from[0]
diff --git a/node_modules/prismarine-viewer/viewer/lib/worker.js b/node_modules/prismarine-viewer/viewer/lib/worker.js
index b743167..eed8b7b 100644
--- a/node_modules/prismarine-viewer/viewer/lib/worker.js
+++ b/node_modules/prismarine-viewer/viewer/lib/worker.js
@@ -29,10 +29,11 @@ function setSectionDirty (pos, value = true) {
   const z = Math.floor(pos.z / 16) * 16
   const chunk = world.getColumn(x, z)
   const key = sectionKey(x, y, z)
+  
   if (!value) {
     delete dirtySections[key]
     postMessage({ type: 'sectionFinished', key })
-  } else if (chunk && chunk.sections[Math.floor(y / 16)]) {
+  } else if (chunk && chunk.sections[Math.floor((y - chunk.minY) / 16)]) {
     dirtySections[key] = value
   } else {
     postMessage({ type: 'sectionFinished', key })
@@ -74,7 +75,7 @@ setInterval(() => {
     y = parseInt(y, 10)
     z = parseInt(z, 10)
     const chunk = world.getColumn(x, z)
-    if (chunk && chunk.sections[Math.floor(y / 16)]) {
+    if (chunk && chunk.sections[Math.floor((y - chunk.minY) / 16)]) {
       delete dirtySections[key]
       const geometry = getSectionGeometry(x, y, z, world, blocksStates)
       const transferable = [geometry.positions.buffer, geometry.normals.buffer, geometry.colors.buffer, geometry.uvs.buffer]
diff --git a/node_modules/prismarine-viewer/viewer/lib/worldrenderer.js b/node_modules/prismarine-viewer/viewer/lib/worldrenderer.js
index 744224a..cf2c60b 100644
--- a/node_modules/prismarine-viewer/viewer/lib/worldrenderer.js
+++ b/node_modules/prismarine-viewer/viewer/lib/worldrenderer.js
@@ -16,6 +16,8 @@ class WorldRenderer {
     this.version = undefined
     this.scene = scene
     this.loadedChunks = {}
+    this.worldMinY = -64
+    this.worldHeight = 384
     this.sectionsOutstanding = new Set()
     this.renderUpdateEmitter = new EventEmitter()
     this.blockStatesData = undefined
@@ -112,7 +114,14 @@ class WorldRenderer {
     for (const worker of this.workers) {
       worker.postMessage({ type: 'chunk', x, z, chunk })
     }
-    for (let y = 0; y < 256; y += 16) {
+    // this is really hacky but ig it works
+    if (this.worldMinY === undefined) {
+      const { minY, worldHeight } = JSON.parse(chunk);
+      this.worldMinY = minY;
+      this.worldHeight = worldHeight;
+    }
+
+    for (let y = this.worldMinY; y < this.worldHeight + this.worldMinY; y += 16) {
       const loc = new Vec3(x, y, z)
       this.setSectionDirty(loc)
       this.setSectionDirty(loc.offset(-16, 0, 0))
@@ -127,7 +136,7 @@ class WorldRenderer {
     for (const worker of this.workers) {
       worker.postMessage({ type: 'unloadChunk', x, z })
     }
-    for (let y = 0; y < 256; y += 16) {
+    for (let y = this.worldMinY; y < this.worldHeight + this.worldMinY; y += 16) { 
       this.setSectionDirty(new Vec3(x, y, z), false)
       const key = `${x},${y},${z}`
       const mesh = this.sectionMeshs[key]
